ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/QueryAnonymizerTest.java

```diff
@@ -0,0 +1,89 @@
+package io.confluent.ksql.test;
+
+import io.confluent.ksql.test.QueryTranslationTest.QttTestFile;
+import io.confluent.ksql.test.loader.JsonTestLoader;
+import io.confluent.ksql.test.tools.TestCase;
+import io.confluent.ksql.util.GrammarTokenExporter;
+import java.io.FileReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+import org.approvaltests.Approvals;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+import io.confluent.ksql.util.QueryAnonymizer;
+import org.junit.Before;
+import org.junit.Test;
+
+public class QueryAnonymizerTest {
+
+  private static final Path QUERIES_TO_ANONYMIZE_PATH =
+      Paths.get("src/test/java/io/confluent/ksql/test/QueriesToAnonymizeTest.txt");
+  private static final Path QUERY_VALIDATION_TEST_DIR = Paths.get("query-validation-tests");
+  private final QueryAnonymizer anon = new QueryAnonymizer();
+  private final Stream<TestCase> testCases = testFileLoader().load();
+  private List<String> sqlTokens;
+
+  @Before
+  public void setUp() {
+    sqlTokens = GrammarTokenExporter.getTokens();
+    sqlTokens.addAll(Arrays.asList("", "INT", "DOUBLE", "VARCHAR", "BOOLEAN", "BIGINT", "VALUE", "KEY"));
+  }
+
+  @Test
+  public void queriesAreAnonymizedCorrectly() throws Exception {
+    // Read queries
+    ArrayList<String> statements = new ArrayList<>();
+    try (Scanner s = new Scanner(new FileReader(QUERIES_TO_ANONYMIZE_PATH.toAbsolutePath().toString()))) {
+      while (s.hasNext()) {
+        statements.add(s.nextLine());
+      }
+    }
+
+    // anonymize statements
+    final List<String> anonStatementSelection = new ArrayList<>();
+    for (String statement : statements) {
+      anonStatementSelection.add(anon.anonymize(statement));
+    }
+
+    // Assert:
+    Approvals.verifyAll("Integration tests", anonStatementSelection);
```

Comment:
just anonymize whole file in one go using the regular approval test patterns? Anonymizer should work whether you give it individual statements or a file full of them separated by semicolons